### ISSUE TITLE
tuple literal types

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#5]
```py
foo: (int, str)
```

**Comment by divaltor** [comment]
I think it's better to stick with PEP standard of typings declaration, because at some moment you can switch to new type checker or use mypy, for example, which couldn't understand such syntax, IMO

**Comment by DetachHead** [comment]
i'm going to put all non-standard features behind an `allowNonStandardFeatures` option i think, that way users can decide for themselves

**Comment by KotlinIsland** [comment]
There is also the scenario of banning non-pep behaviour for public API/evaluated extressions

**Comment by DetachHead** [comment]
we need to also account for their usage inside generics, because these two mean the same thing at runtime:

```py
foo: list[(int, str)]
bar: list[int, str]
```

the only way to make it work is to add a trailing comma, which looks god awful:

```py
foo: list[(int, str),]
```

so i think the only possible solutions are either:

1. ban tuple literal syntax inside generics
2. enforce `__future__.annotations` and just treat `list[(int, str)]` as a list of tuples, and `list[int, str]` as an invalid type

**Comment by KotlinIsland** [comment]
additionally, generic constraints are now provided with a tuple:
```py
def f[T: (int, str)](t: T): ...
```

**Comment by DetachHead** [comment]
i dont think this is feasible unless we can come up with a solution to both of those problems

**Comment by KotlinIsland** [comment]
with the ones in generics, just don't allow it for now, until a more comprehensive solution can be built. 
for constraints, just take it as a constraint.

**Comment by DetachHead** [comment]
i don't like the idea of changing the syntax to something that means something else in certain contexts, and is only sometimes allowed. this will just add more confusion to the type system which is already a complete mess
